File: src/main/java/net/dontdrinkandroot/example/angularrestspringsecurity/fixtures/AutowiringFixtureLoader.java
package net.dontdrinkandroot.example.angularrestspringsecurity.fixtures;

import net.dontdrinkandroot.fixtures.Fixture;
import net.dontdrinkandroot.fixtures.loader.DefaultFixtureLoader;
import net.dontdrinkandroot.fixtures.purger.DatabasePurger;
import org.springframework.beans.factory.config.AutowireCapableBeanFactory;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class AutowiringFixtureLoader extends DefaultFixtureLoader
{
    private AutowireCapableBeanFactory beanFactory;

    public AutowiringFixtureLoader(DatabasePurger databasePurger, AutowireCapableBeanFactory beanFactory)
    {
        super(databasePurger);
        this.beanFactory = beanFactory;
    }

    @Override
    protected Fixture instantiateFixtureClass(Class<? extends Fixture> fixtureClass)
    {
        return (Fixture) this.beanFactory.autowire(
                fixtureClass,
                AutowireCapableBeanFactory.AUTOWIRE_CONSTRUCTOR,
                false
        );
    }
}

File: src/main/kotlin/net/dontdrinkandroot/example/angularrestspringsecurity/fixtures/AutowiringFixtureLoader.kt
package net.dontdrinkandroot.example.angularrestspringsecurity.fixtures

import net.dontdrinkandroot.fixtures.Fixture
import net.dontdrinkandroot.fixtures.loader.DefaultFixtureLoader
import net.dontdrinkandroot.fixtures.purger.DatabasePurger
import org.springframework.beans.factory.config.AutowireCapableBeanFactory

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
open class AutowiringFixtureLoader(
        databasePurger: DatabasePurger,
        private val beanFactory: AutowireCapableBeanFactory
) : DefaultFixtureLoader(databasePurger) {

    @Throws(InstantiationException::class, IllegalAccessException::class)
    override fun instantiateFixtureClass(fixtureClass: Class<out Fixture>): Fixture {
        return beanFactory.autowire(
                fixtureClass,
                AutowireCapableBeanFactory.AUTOWIRE_CONSTRUCTOR,
                false
        ) as Fixture
    }
}
